controllers/managedfusion_controller.go

```diff
@@ -689,6 +723,158 @@ func (r *ManagedFusionReconciler) reconcilePrometheusProxyNetworkPolicy() error
 	return nil
 }
 
+func (r *ManagedFusionReconciler) reconcileEgressFirewall() error {
+	if !r.AvailableCRDs[EgressFirewallCRD] {
+		r.Log.Info("EgressFirewall CRD not present, skipping reconcile for egress firewall")
+		return nil
+	}
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.egressFirewall, func() error {
+		if err := r.own(r.egressFirewall); err != nil {
+			return err
+		}
+		desired := templates.EgressFirewallTemplate.DeepCopy()
+
+		// Get the aws config map
+		awsConfigMap := &corev1.ConfigMap{}
+		awsConfigMap.Name = utils.IMDSConfigMapName
+		awsConfigMap.Namespace = r.Namespace
+		if err := r.get(awsConfigMap); err != nil {
+			return fmt.Errorf("unable to get AWS ConfigMap: %v", err)
+		}
+
+		// Get the machine cidr
+		vpcCIDR, ok := awsConfigMap.Data[utils.CIDRKey]
+		if !ok {
+			return fmt.Errorf("unable to determine machine CIDR from AWS ConfigMap")
+		}
+		// Allow egress traffic to that cidr range
+		var vpcEgressRules []ovnv1.EgressFirewallRule
+		cidrList := strings.Split(vpcCIDR, ";")
+		for _, cidr := range cidrList {
+			if cidr == "" {
+				continue
+			}
+			vpcEgressRules = append(vpcEgressRules, ovnv1.EgressFirewallRule{
+				Type: ovnv1.EgressFirewallRuleAllow,
+				To: ovnv1.EgressFirewallDestination{
+					CIDRSelector: cidr,
+				},
+			})
+		}
+
+		// Inserting the VPC Egress rule in front of all other egress rules.
+		// The order or rules matter
+		// https://docs.openshift.com/container-platform/4.10/networking/openshift_sdn/configuring-egress-firewall.html#policy-rule-order_openshift-sdn-egress-firewall
+		// Inserting this rule in the front ensures it comes before the EgressNetworkPolicyRuleDeny rule.
+		desired.Spec.Egress = append(
+			vpcEgressRules,
+			desired.Spec.Egress...,
+		)
+
+		smtpURL, err := url.Parse(r.smtpConfigData.Endpoint)
```

Comment:
- if the endpoint should strictly be a url with protocol included ex: `http://example.com` vs `example.com`, it's better to use `url.ParseRequestURI` rather than `url.Parse`
- In general `url.Parse` is very loosely validated

Reply:
Updated